src/misc/lv_array.c

```diff
@@ -75,15 +75,21 @@ lv_result_t lv_array_remove(lv_array_t * array, uint32_t index)
 
     /*Shortcut*/
     if(index == array->size - 1) {
-        lv_array_resize(array, array->size - 1);
+        array->size--;
```

Comment:
Let's add a static function and use it elsewhere.
```c
lv_array_shrink(array, newsize) {
   if (newsize < array->size && newsize <= array->capacity >> 1) {
      lv_array_resize(array, newsize);
   }
}
```

Reply:
I agree that this new functionality should be in a separate function.


... and I wonder if it should be behind a toggle: 
- by default resize array to half
- but on devices with much less memory keep the original resize logic (AKA resize immediately), because keeping many of these "blown up" arrays in memory without decreasing them could "eat up memory".
This is only an assumption if there are too many `lv_array`s used... but we would see clearer picture if it's really measured (number of arrays, bytes used / allocated (fill ratio) through time, etc)

Reply:
> I agree that this new functionality should be in a separate function.
> 
> ... and I wonder if it should be behind a toggle:
> 
> * by default resize array to half when fill ratio decreases to 50%
> * but on devices with much less memory keep the original resize logic (AKA resize immediately), because keeping many of these "blown up" arrays in memory without decreasing them could "eat up memory".
>   This is only an assumption if there are too many `lv_array`s used... but we would see clearer picture if it's really measured (number of arrays, bytes used / capacity (fill ratio) through time, etc)

      I disagree with this perspective. Even on devices with limited memory, it is not advisable to frequently use the resize function. This can result in higher memory fragmentation, leading to faster memory depletion and significantly impacting overall performance.
      I agree that the size of this resize could be made configurable to quickly free up memory on lower-end devices, for example config it to 30% or 20 %.